test/common_utils.py

```diff
@@ -14,3 +20,135 @@ def get_tmp_dir(src=None, **kwargs):
+for i, arg in enumerate(sys.argv):
+    if arg == '--accept':
+        del sys.argv[i]
+        break
+
+
+class MapNestedTensorObjectImpl(object):
+    def __init__(self, tensor_map_fn):
+        self.tensor_map_fn = tensor_map_fn
+
+    def __call__(self, object):
+        if isinstance(object, torch.Tensor):
+            return self.tensor_map_fn(object)
+
+        elif isinstance(object, dict):
+            mapped_dict = {}
+            for key, value in object.items():
+                mapped_dict[self(key)] = self(value)
+            return mapped_dict
+
+        elif isinstance(object, (list, tuple)):
+            mapped_iter = []
+            for iter in object:
+                mapped_iter.append(self(iter))
+            return mapped_iter if not isinstance(object, tuple) else tuple(mapped_iter)
+
+        else:
+            return object
+
+
+def map_nested_tensor_object(object, tensor_map_fn):
+    impl = MapNestedTensorObjectImpl(tensor_map_fn)
+    return impl(object)
+
+
+# adapted from TestCase in torch/test/common_utils
+class TestCase(unittest.TestCase):
+    def assertExpected(self, output, subname=None, rtol=None, atol=None):
+        r"""
+        Test that a python value matches the recorded contents of a file
+        derived from the name of this test and subname.  The value must be
+        pickable with `torch.load`. This file
+        is placed in the 'expect' directory in the same directory
+        as the test script. You can automatically update the recorded test
+        output using --accept.
+
+        If you call this multiple times in a single function, you must
+        give a unique subname each time.
+        """
+        def remove_prefix(text, prefix):
+            if text.startswith(prefix):
+                return text[len(prefix):]
+            return text
+        # NB: we take __file__ from the module that defined the test
+        # class, so we place the expect directory where the test script
+        # lives, NOT where test/common_utils.py lives.  This doesn't matter in
+        # PyTorch where all test scripts are in the same directory as
+        # test/common_utils.py, but it matters in onnx-pytorch
+        module_id = self.__class__.__module__
+        munged_id = remove_prefix(self.id(), module_id + ".")
+        test_file = os.path.realpath(sys.modules[module_id].__file__)
+        expected_file = os.path.join(os.path.dirname(test_file),
+                                     "expect",
+                                     munged_id)
+
+        subname_output = ""
+        if subname:
+            expected_file += "_" + subname
+            subname_output = " ({})".format(subname)
+        expected_file += "_expect.pkl"
+        expected = None
+
+        def accept_output(update_type):
+            print("Accepting {} for {}{}:\n\n{}".format(update_type, munged_id, subname_output, output))
+            torch.save(output, expected_file)
+            MAX_PICKLE_SIZE = 100 * 1000  # 100 KB
```

Comment:
Cool, I think 100KB is ok as an upper limit, although twice as much as the largest (text) file we currently have in torchvision.